mongo/errors.go

```diff
@@ -202,13 +202,13 @@ type ServerError interface {
 	HasErrorMessage(string) bool
 	// HasErrorCodeWithMessage returns true if any of the contained errors have the specified code and message.
 	HasErrorCodeWithMessage(int, string) bool
-
-	serverError()
```

Comment:
Implementing `ServerError` for types outside of the `mongo` package (in this case in `driver`) means that I must remove this guard. Without `serverError()`, users _can_ now implement the `ServerError` interface for their own types. We do [document](https://github.com/mongodb/mongo-go-driver/blob/master/mongo/errors.go#L193-L194) that this is discouraged, but if there's some way I'm missing to keep this guard in, let me know 🌺

Reply:
Hmm, I don't love the repetition, but it might be better than removing the guard altogether. Any thoughts @kevinAlbs?

Reply:
@iwysiu does copying the interface take away from the usability?

I thought the motivation of the `ServerError` interface was to be a convenient interface to inspect all server errors.

```
_, err = client.Database("db").Collection("coll").InsertOne(ctx, bson.D{{"_id", "foo"}})
	if err != nil {
		fmt.Println("Insert failed", err)
		var serverError mongo.ServerError

		if errors.As(err, &serverError) {
			if serverError.HasErrorLabel("RetryableWrite") {
				fmt.Println("Going to retry...")
			}
		}
	}
```

A distinct `driver.ServerError` would be less accessible if it is in the `x` package, and it would require the user to check for both interfaces.

Reply:
It would make it more annoying for end users. If we're not worried about users implementing it, i think removing the guard is fine

Reply:
I don't think there's much risk to "unsealing" the `ServerError` interface. As @kevinAlbs mentioned, nothing uses it as an argument directly, and even if someone implemented the interface, they wouldn't get any unexpected behavior, it just doesn't make make sense to do so.

Alternately, there's actually no reason that `ServerError` needs to be exported. We could leave `ServerError` as is, define a new unexported interface like `serverError` with the same function signatures,  add a note to `ServerError` that it's unused and only there to maintain public API compatibility.